src/test/api.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { expect } from "chai";
+
+import * as utils from "../utils";
+
+describe("api", () => {
+  beforeEach(() => {
+    // The api module resolves env var statically so we need to
+    // do lazy imports and clear the import each time.
+    delete require.cache[require.resolve("../api")];
+  });
+
+  afterEach(() => {
+    delete process.env.FIREBASE_DATABASE_EMULATOR_HOST;
+    delete process.env.FIREBASE_REALTIME_URL;
+    delete process.env.FIRESTORE_EMULATOR_HOST;
+    delete process.env.FIRESTORE_URL;
+
+    // This is dirty, but utils keeps stateful overrides and we need to clear it
+    utils.envOverrides.splice(0, utils.envOverrides.length);
```

Comment:
What does this actually do? It's not clear to me what you're trying to achieve on a lower level.

If you want to clear the array in place, use `utils.envOverrides.length = 0`.

Reply:
That's what I want to do, I will use your method.